Prometheus module does not handle spaces in label values

The Prometheus metrics collector does not appear to handle spaces in the label values. For example, this metric:
```
jvm_memory_committed_bytes{area="nonheap",id="Code Cache",} 6.1997056E7
jvm_memory_committed_bytes{area="nonheap",id="Compressed Class Space",} 1.1927552E7
jvm_memory_committed_bytes{area="heap",id="Eden Space",} 2.86326784E8
jvm_memory_committed_bytes{area="heap",id="Survivor Space",} 3.5782656E7
jvm_memory_committed_bytes{area="heap",id="Tenured Gen",} 7.15849728E8
```

Will cause the following exception:
```
[ WARN  ] prometheus[prometheus] collect_gauge_counter.go:41 error on adding dim to chart 'jvm_gc_pause_seconds_count' : unacceptable symbol in dim ID 'jvm_gc_pause_seconds_count|action=end of minor GC,cause=Allocation Failure' : ' '
[ WARN  ] prometheus[prometheus] collect_gauge_counter.go:41 error on adding dim to chart 'jvm_gc_pause_seconds_count' : unacceptable symbol in dim ID 'jvm_gc_pause_seconds_count|action=end of major GC,cause=Metadata GC Threshold' : ' '
[ WARN  ] prometheus[prometheus] collect_gauge_counter.go:41 error on adding dim to chart 'jvm_gc_pause_seconds_max' : unacceptable symbol in dim ID 'jvm_gc_pause_seconds_max|action=end of minor GC,cause=Allocation Failure' : ' '
[ WARN  ] prometheus[prometheus] collect_gauge_counter.go:41 error on adding dim to chart 'jvm_gc_pause_seconds_max' : unacceptable symbol in dim ID 'jvm_gc_pause_seconds_max|action=end of major GC,cause=Metadata GC Threshold' : ' '
[ WARN  ] prometheus[prometheus] collect_gauge_counter.go:41 error on adding dim to chart 'jvm_gc_pause_seconds_sum' : unacceptable symbol in dim ID 'jvm_gc_pause_seconds_sum|action=end of major GC,cause=Metadata GC Threshold' : ' '
[ WARN  ] prometheus[prometheus] collect_gauge_counter.go:41 error on adding dim to chart 'jvm_gc_pause_seconds_sum' : unacceptable symbol in dim ID 'jvm_gc_pause_seconds_sum|action=end of minor GC,cause=Allocation Failure' : ' '
[ WARN  ] prometheus[prometheus] collect_gauge_counter.go:41 error on adding dim to chart 'jvm_memory_committed_bytes' : unacceptable symbol in dim ID 'jvm_memory_committed_bytes|area=nonheap,id=Compressed Class Space' : ' '
[ WARN  ] prometheus[prometheus] collect_gauge_counter.go:41 error on adding dim to chart 'jvm_memory_committed_bytes' : unacceptable symbol in dim ID 'jvm_memory_committed_bytes|area=heap,id=Tenured Gen' : ' '
[ WARN  ] prometheus[prometheus] collect_gauge_counter.go:41 error on adding dim to chart 'jvm_memory_committed_bytes' : unacceptable symbol in dim ID 'jvm_memory_committed_bytes|area=heap,id=Eden Space' : ' '
[ WARN  ] prometheus[prometheus] collect_gauge_counter.go:41 error on adding dim to chart 'jvm_memory_committed_bytes' : unacceptable symbol in dim ID 'jvm_memory_committed_bytes|area=nonheap,id=Code Cache' : ' '
[ WARN  ] prometheus[prometheus] collect_gauge_counter.go:41 error on adding dim to chart 'jvm_memory_committed_bytes' : unacceptable symbol in dim ID 'jvm_memory_committed_bytes|area=heap,id=Survivor Space' : ' '
[ WARN  ] prometheus[prometheus] collect_gauge_counter.go:41 error on adding dim to chart 'jvm_memory_max_bytes' : unacceptable symbol in dim ID 'jvm_memory_max_bytes|area=heap,id=Survivor Space' : ' '
[ WARN  ] prometheus[prometheus] collect_gauge_counter.go:41 error on adding dim to chart 'jvm_memory_max_bytes' : unacceptable symbol in dim ID 'jvm_memory_max_bytes|area=nonheap,id=Compressed Class Space' : ' '
[ WARN  ] prometheus[prometheus] collect_gauge_counter.go:41 error on adding dim to chart 'jvm_memory_max_bytes' : unacceptable symbol in dim ID 'jvm_memory_max_bytes|area=heap,id=Tenured Gen' : ' '
[ WARN  ] prometheus[prometheus] collect_gauge_counter.go:41 error on adding dim to chart 'jvm_memory_max_bytes' : unacceptable symbol in dim ID 'jvm_memory_max_bytes|area=heap,id=Eden Space' : ' '
[ WARN  ] prometheus[prometheus] collect_gauge_counter.go:41 error on adding dim to chart 'jvm_memory_max_bytes' : unacceptable symbol in dim ID 'jvm_memory_max_bytes|area=nonheap,id=Code Cache' : ' '
[ WARN  ] prometheus[prometheus] collect_gauge_counter.go:41 error on adding dim to chart 'jvm_memory_used_bytes' : unacceptable symbol in dim ID 'jvm_memory_used_bytes|area=heap,id=Tenured Gen' : ' '
[ WARN  ] prometheus[prometheus] collect_gauge_counter.go:41 error on adding dim to chart 'jvm_memory_used_bytes' : unacceptable symbol in dim ID 'jvm_memory_used_bytes|area=nonheap,id=Code Cache' : ' '
[ WARN  ] prometheus[prometheus] collect_gauge_counter.go:41 error on adding dim to chart 'jvm_memory_used_bytes' : unacceptable symbol in dim ID 'jvm_memory_used_bytes|area=heap,id=Eden Space' : ' '
[ WARN  ] prometheus[prometheus] collect_gauge_counter.go:41 error on adding dim to chart 'jvm_memory_used_bytes' : unacceptable symbol in dim ID 'jvm_memory_used_bytes|area=heap,id=Survivor Space' : ' '
[ WARN  ] prometheus[prometheus] collect_gauge_counter.go:41 error on adding dim to chart 'jvm_memory_used_bytes' : unacceptable symbol in dim ID 'jvm_memory_used_bytes|area=nonheap,id=Compressed Class Space' : ' '
```

--- a/modules/prometheus/group.go
+++ b/modules/prometheus/group.go
@@ -120,15 +120,17 @@ loop:
 		comma = true
 		id.WriteString(label.Name)
 		id.WriteString("=")
-		if strings.IndexByte(label.Value, '\\') != -1 {
-			if v := decodeLabelValue(label.Value); strings.IndexByte(v, '\\') != -1 {
-				id.WriteString(labelValueReplacer.Replace(v))
-			} else {
-				id.WriteString(v)
-			}
-		} else {
-			id.WriteString(label.Value)
+
+		lval := label.Value
+		if strings.IndexByte(lval, ' ') != -1 {
+			lval = spaceReplacer.Replace(lval)
 		}
+		if strings.IndexByte(lval, '\\') != -1 {
+			if lval = decodeLabelValue(lval); strings.IndexByte(lval, '\\') != -1 {
+				lval = backslashReplacer.Replace(lval)
+			}
+		}
+		id.WriteString(lval)
 	}
 	return id.String()
 }
@@ -153,6 +155,7 @@ func contains(in []string, value string) bool {
 }
 
 // TODO: fix
-var labelValueReplacer = strings.NewReplacer(
-	`\`, "_",
+var (
+	spaceReplacer     = strings.NewReplacer(" ", "_")
+	backslashReplacer = strings.NewReplacer(`\`, "_")
 )

--- a/modules/prometheus/prometheus_test.go
+++ b/modules/prometheus/prometheus_test.go
@@ -7,11 +7,11 @@ import (
 	"strings"
 	"testing"
 
+	"github.com/netdata/go.d.plugin/agent/module"
 	"github.com/netdata/go.d.plugin/pkg/prometheus/selector"
 	"github.com/netdata/go.d.plugin/pkg/tlscfg"
 	"github.com/netdata/go.d.plugin/pkg/web"
 
-	"github.com/netdata/go.d.plugin/agent/module"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
@@ -390,6 +390,20 @@ func TestPrometheus_Collect(t *testing.T) {
 			},
 			wantCollected: map[string]int64{
 				"http_time_to_write_seconds|code=200,route=^/([^/]+/){1,}[^/]+/uploads_z": 0,
+
+				"series":  1,
+				"metrics": 1,
+				"charts":  int64(1 + len(statsCharts)),
+			},
+		},
+		`label value contains ' '`: {
+			input: [][]string{
+				{
+					`jvm_memory_max_bytes{area="heap",id="Eden Space"} 0`,
+				},
+			},
+			wantCollected: map[string]int64{
+				"jvm_memory_max_bytes|area=heap,id=Eden_Space": 0,
 
 				"series":  1,
 				"metrics": 1,

